### PR TITLE
[FLINK-4263] [table] SQL's VALUES does not work properly

### PR DESCRIPTION
**Pull request by twalthr**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This PR reworks the creation of SQL VALUES. It uses code generation and creates the rows within the input format, thus, we don't have to care about serializability of literals or rows. I updated the Janino compiler version to solve some compilation problems.

### CODE REVIEW

File: flink-libraries/flink-table/pom.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@ under the License.
 		<dependency>
 			<groupId>org.codehaus.janino</groupId>
 			<artifactId>janino</artifactId>
-			<version>2.7.5</version>
+			<version>3.0.6</version>
```
Comment (fhueske) [inline flink-libraries/flink-table/pom.xml]: Have you checked that the transitive dependencies of Janino 3.0.6 are OK?

**Comment by fhueske** [comment]
Code looks good. We should check the updated Janino dependency though.
+1 to merge if that is OK.

File: flink-libraries/flink-table/pom.xml (hunk 1 above)
Comment (ex00) [inline flink-libraries/flink-table/pom.xml]: Hi
I checked transitive dependencies in maven. I used `dependency:tree` plugin
for janino:2.7.5 is:
```
org.codehaus.janino:janino:jar:2.7.5
+- junit:junit:jar:4.7:test
+- org.apache.ant:ant-nodeps:jar:1.7.1:compile
|  \- org.apache.ant:ant:jar:1.7.1:compile
|     \- org.apache.ant:ant-launcher:jar:1.7.1:compile
\- org.codehaus.janino:commons-compiler:jar:2.7.5:compile
```
for janino:3.0.6 is:
```
org.codehaus.janino:janino:jar:3.0.6
+- junit:junit:jar:4.12:test
|  \- org.hamcrest:hamcrest-core:jar:1.3:test
+- org.codehaus.janino:commons-compiler:jar:3.0.6:compile
\- org.apache.ant:ant:jar:1.7.1:provided
   \- org.apache.ant:ant-launcher:jar:1.7.1:provided
```

**Comment by twalthr** [comment]
Thanks @fhueske and @ex00. I will merge this.

**Pull request closed without merging**